server/src/main/java/org/opensearch/cluster/coordination/JoinHelper.java

```diff
@@ -122,6 +131,8 @@ public class JoinHelper {
 
     private final Supplier<JoinTaskExecutor> joinTaskExecutorGenerator;
     private final Consumer<Boolean> nodeCommissioned;
+    private final NamedWriteableRegistry namedWriteableRegistry;
+    private final Cache<Long, BytesReference> serializedStates;
```

Comment:
Do we need to use a `Cache` here? Would a `HashMap` or `ConcurrentHashMap` be sufficient?

Reply:
Yes! Normal ConcurrentHashMap is good enough. Changing it back to CHM.